src/plugins/discover/public/application/components/context_app/context_app_legacy.tsx

```diff
@@ -41,17 +51,20 @@ export interface ContextAppProps {
   useNewFieldsApi?: boolean;
 }
 
+const DataGridMemoized = React.memo(DiscoverGrid);
 const PREDECESSOR_TYPE = 'predecessors';
 const SUCCESSOR_TYPE = 'successors';
 
 function isLoading(status: string) {
-  return status !== LOADING_STATUS.LOADED && status !== LOADING_STATUS.FAILED;
+  return status !== LoadingStatus.LOADED && status !== LoadingStatus.FAILED;
 }
 
 export function ContextAppLegacy(renderProps: ContextAppProps) {
+  const [expandedDoc, setExpandedDoc] = useState<ElasticSearchHit | undefined>(undefined);
```

Comment:
since the container controller will go away , you could call 

```
const services = getServices();
```
here, so no need to extend the legacy code, which will go away anyway